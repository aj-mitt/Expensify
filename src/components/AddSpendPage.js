import React from 'react'
import {connect} from 'react-redux'
import SpendingForm from './SpendingForm'
import {startAddSpending} from '../actions/spending'

export class AddSpendPage extends React.Component {
  onSubmit = (spending) => {
    this.props.startAddSpending(spending)
    this.props.history.push('/')
  }
  render() {
    return(
      <div>
      <h1>Add Spending</h1>
      <SpendingForm
        onSubmit={this.onSubmit}
      />
    </div>
      
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddSpending: (spending) => dispatch(startAddSpending(spending))
})

export default connect(undefined, mapDispatchToProps)(AddSpendPage)
