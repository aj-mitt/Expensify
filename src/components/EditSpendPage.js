import React from 'react'
import {connect} from 'react-redux'
import SpendingForm from './SpendingForm'
import { startEditSpending, startRemoveSpending } from '../actions/spending';

export class EditSpendPage extends React.Component {
  onSubmit = (spending) => {
    this.props.startEditSpending(this.props.spending.id, spending)
    this.props.history.push('/')
  }
  onRemove = () => {
    this.props.startRemoveSpending({id: this.props.spending.id})
    this.props.history.push('/')
  }
  render(){
    return (
      <div>
        <SpendingForm
          spending={this.props.spending}
          onSubmit= {this.onSubmit}
        />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => {
  return {
    spending: state.spending.find((spending) => spending.id === props.match.params.id)
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  startEditSpending: (id, spending) => dispatch(startEditSpending(id, spending)),
  startRemoveSpending: (data) => dispatch(startRemoveSpending(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditSpendPage)
