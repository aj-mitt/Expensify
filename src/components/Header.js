import React from 'react'
import {NavLink} from 'react-router-dom'
import {connect} from 'react-redux'
import {startLogout} from '../actions/auth'

export const Header = ({startLogout}) => (
  <header>
    <h1>Spend Track</h1>
    <NavLink to='/dashborad' activeClassName='is-active'>Home</NavLink>
    <NavLink to='/create' activeClassName='is-active'>Create Spend</NavLink>
    <button onClick={startLogout}>Logout</button>
  </header>
)
const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)
