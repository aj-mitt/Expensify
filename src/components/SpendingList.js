import React from 'react'
import {connect} from 'react-redux'
import SpendingListItem from './SpendingListItem'
import selectSpending from '../selectors/spending'

export const SpendingList = (props) => (
  <div>
    {
      props.spending.length === 0 ? (
        <p> No Spending</p>
      ) : (
        props.spending.map((spending) => {
          return <SpendingListItem key={spending.id} {...spending}/>
        })
      )
    }
  </div>
)

const mapStateToProps = (state) => {
  return {
    spending: selectSpending(state.spending, state.filters)
  }
}

export default connect(mapStateToProps)(SpendingList)
