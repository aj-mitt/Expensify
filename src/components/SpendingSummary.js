import React from 'react'
import {connect} from 'react-redux'
import numeral from 'numeral'
import selectSpending from '../selectors/spending'
import selectSpendingTotal from '../selectors/spending-total'

export const SpendingSummary = ({spendingCount, spendingTotal}) => {
  const spendingWord = spendingCount === 1 ? 'spending' : 'spending'
  const formattedSpendingTotal = numeral(spendingTotal / 100).format('$0,0.00')
  return (
    <div>
      <h1>Viewing {spendingCount} {spendingWord} total  of {formattedSpendingTotal}</h1>
    </div>
  )
}

const mapStateToProps = (state) => {
  const visibleSpending = selectSpending(state.spending, state.filters)

  return {
    spendingCount: visibleSpending.length,
    spendingTotal: selectSpendingTotal(visibleSpending)
  }
}

export default connect(mapStateToProps)(SpendingSummary)