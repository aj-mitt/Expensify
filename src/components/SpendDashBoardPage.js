import React from 'react'
import SpendingList from './SpendingList'
import SpendingListFilters from './SpendingListFilters'
import SpendingSummary from './SpendingSummary'

const SpendDashboardPage = () => (
  <div>
    <SpendingSummary />
    <SpendingListFilters />
    <SpendingList />
  </div>

)

export default SpendDashboardPage
