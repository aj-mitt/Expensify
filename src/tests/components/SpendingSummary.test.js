import React from 'react'
import {shallow} from 'enzyme'
import {SpendingSummary} from '../../components/SpendingSummary'

test('Should test render spending summary with 1 spending', () => {
  const wrapper = shallow(<SpendingSummary spendingCount={1} spendingTotal={235} />)
  expect(wrapper).toMatchSnapshot()
})

test('Should test render spending summary with multiply spending', () => {
  const wrapper = shallow(<SpendingSummary spendingCount={23} spendingTotal={23597796167} />)
  expect(wrapper).toMatchSnapshot()
})
