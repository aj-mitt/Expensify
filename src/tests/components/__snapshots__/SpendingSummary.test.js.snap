// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test render spending summary with 1 spending 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    spending
     total  of 
    $2.35
  </h1>
</div>
`;

exports[`Should test render spending summary with multiply spending 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    spending
     total  of 
    $235,977,961.67
  </h1>
</div>
`;
